crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -1304,6 +1336,18 @@ impl<'db> Parameter<'db> {
         self
     }
 
+    fn materialize(&self, db: &'db dyn Db, variance: TypeVarVariance) -> Self {
+        Self {
+            annotated_type: Some(
+                self.annotated_type
+                    .unwrap_or(Type::unknown())
+                    .materialize(db, variance.flip()),
```

Comment:
Probably doesn't matter, but we could do this flip just once on line 368 instead of separately for each parameter?